index.js

```diff
@@ -105,6 +115,16 @@ Funnel.prototype.build = function() {
     inputPath = path.join(inputPath, this.srcDir);
   }
 
+  if (this._dynamicFilesFunc) {
+    this.lastFiles = this.files;
+    this.files = this._dynamicFilesFunc() || [];
```

Comment:
I don't know if a sample-size of 1 (myself) is enough to say :).

But to guess, I could see it being common to use a `files` func that always returns the same array (in the situation where the array is static but you simply don't have it ready-to-go at `Brocfile.js` definition-time). Also, it seemed like a pretty trivial, yet significant, optimization so I went for it.